### PR TITLE
VIH-9730 Editing-non-availability-hours-filter-by-dates-not-populating-particular-results

### PR DESCRIPTION
**Pull request by arellison**
### **JIRA link** 

https://tools.hmcts.net/jira/browse/VIH-9730

### **Change description**
When filtering and saving the non-availability hours by date in the ‘Manage work allocation’ section, the date filter form is reset, and the data returns all results rather than just those for the dates specified by the current date filter.

To solve this problem, an isFiltered boolean has been added and is set to True if the non-availability date filter form has been submitted. This means that if the isFiltered boolean is set to True while saving non-availability hours, the non-availability date filter is re-run and the correct results are returned.

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts [hunk 1]
```diff
@@ -39,12 +39,19 @@ export class VhoWorkHoursNonAvailabilityTableComponent implements OnInit, CanDea
...
+            this.resetStartDateAndEndDate();
+        }
         this.hideMessage();
         if (value) {
             this.nonAvailabilityWorkHoursResponses = value;
             this.nonWorkHours = value.map(x => this.mapNonWorkingHoursToEditModel(x));
             this.nonWorkHours = this.nonWorkHours.slice(0, this.filterSize);
+            //if the form has has already been filtered, run-run the filter again
+            if (this.isFiltered) {
+                this.filterByDate();
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts:53]: can we add a test to assert that the date filter is applied here? If you check the sonar cloud coverage, you'll see that this line isn't covered by tests

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts [hunk 2]
```diff
@@ -89,6 +96,7 @@ export class VhoWorkHoursNonAvailabilityTableComponent implements OnInit, CanDea
     originalNonWorkHours: EditVhoNonAvailabilityWorkHoursModel[];
     isEditing = false;
     isSaving = false;
+    isFiltered = false; //form is not filtered by default
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts:99]: I don't think these comments are necessary, it's already quite clear what the code is doing.

**Review by oliver-scott — COMMENTED** [review]
looks good, have added a couple of suggestions

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts (hunk 1 above)
Comment (arellison) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts:53]: @oliver-scott I've added a test that will cover these additional lines.

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts (hunk 2 above)
Comment (oliver-scott) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts]: sorry @arellison I should have been clearer - I think all those comments can go. They don't really add very much

Comment (arellison) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts]: I've removed all the comments now @oliver-scott

Comment (oliver-scott) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts]: great, just a linting issue to fix. `npm run fix-lint`, commit and push should do it :)

Comment (arellison) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts]: I've ran npm run fix-lint, committed and pushed now @oliver-scott

**Review by oliver-scott — APPROVED** [review]

**Review by will-craig — APPROVED** [review]

**Review by twyfordr273 — APPROVED** [review]
Approved

**Pull request merged**